utils/version.go

```diff
@@ -16,5 +16,65 @@ limitations under the License.
 
 package utils
 
+import (
+	"errors"
+	"fmt"
+	"strconv"
+	"strings"
+)
+
 // VERSION is current Acra suite version
-var VERSION = "0.84.2" // change on current during build
+// store it as string instead initialized struct value to easy change/grep/sed/replace value via scripts or with
+// -ldflags "-X github.com/cossacklabs/acra/utils.VERSION=X.X.X"
+var VERSION = "0.84.2"
+
+// Version store version info
+type Version struct {
+	Major string
+	Minor string
+	Patch string
+}
+
+// String format version as string
+func (v *Version) String() string {
+	return fmt.Sprintf("%s.%s.%s", v.Major, v.Minor, v.Patch)
+}
+
+// MajorAsFloat64 return major number as float64
+func (v *Version) MajorAsFloat64() (float64, error) {
```

Comment:
Version components represented as _real_ numbers look very weird to me. Using strings for components storage is weird as well.

I think it would be more natural to store `ints` in `Version`. We validate them as integers in `GetParsedVersion` after all. Then we can cast them as `float64` only in `ExportVersionMetric` as using real numbers looks like an implementation detail of gauges to me.

Reply:
version used only in two places: as metrics where they exported as float64 according to prometheus client interface and as string in log message at the start
as is from `VERSION` variable. Storing as string in `VERSION` var allow override it at compilation time with command `go build -ldflags '-X github.com/cossacklabs/acra/utils.VERSION=<my_new_version>'` via deploying tools with hash of commit or other build's metadata. Golang allow override only string values, so I decided to store it as a string and parse into the structure for metrics or other usages.

in a future version may be extended with pre-release part or with build metadata as it allowed in [semver specification](https://semver.org/). and I thought that it is okay to store as strings and provide getters for specific types if necessary.